test/TPP/transform/transform-convolutions.mlir

```diff
@@ -285,3 +290,60 @@ transform.sequence failures(propagate) {
+  // CHECK-DAG: %[[C32:.+]] = arith.constant 32 : index
+  %0 = tensor.empty() : tensor<1x56x56x256xf32>
+  // CHECK: %{{.*}} = scf.for %{{.*}} = %[[C0]] to %[[C8]] step %[[C1]] iter_args
+  // CHECK: %{{.*}} = scf.for %{{.*}} = %[[C0]] to %[[C56]] step %[[C1]] iter_args
+  // CHECK: %{{.*}} = scf.for %{{.*}} = %[[C0]] to %[[C56]] step %[[C1]] iter_args
+  // CHECK: %{{.*}} = scf.for %{{.*}} = %[[C0]] to %[[C32]] step %[[C1]] iter_args
+  // CHECK: %{{.*}} = scf.for %{{.*}} = %[[C0]] to %[[C2]] step %[[C1]] iter_args
+  // CHECK-COUNT-2: %{{.*}} = linalg.matmul ins(%{{.*}}, %{{.*}} : tensor<1x32xf32>, tensor<32x1xf32>) outs(%{{.*}} : tensor<1x1xf32>) -> tensor<1x1xf32>
+  // CHECK: %{{.*}} = linalg.generic {{.*}}, iterator_types = ["parallel", "parallel", "parallel", "parallel", "parallel"], library_call = "tpp.add"} ins(%{{.*}} : tensor<1x1x1x1x1xf32>) outs(%{{.*}} : tensor<1x1x1x1x1xf32>)
+  %2 = linalg.conv_2d_nhwc_hwcf {dilations = dense<1> : tensor<2xi64>, strides = dense<1> : tensor<2xi64>} ins(%arg0, %arg1 : tensor<1x56x56x64xf32>, tensor<1x1x64x256xf32>) outs(%0 : tensor<1x56x56x256xf32>) -> tensor<1x56x56x256xf32>
+  %3 = linalg.conv_2d_nhwc_hwcf {dilations = dense<1> : tensor<2xi64>, strides = dense<1> : tensor<2xi64>} ins(%arg2, %arg3 : tensor<1x56x56x64xf32>, tensor<1x1x64x256xf32>) outs(%1 : tensor<1x56x56x256xf32>) -> tensor<1x56x56x256xf32>
+  %4 = linalg.generic {indexing_maps = [#map, #map], iterator_types = ["parallel", "parallel", "parallel", "parallel"]} ins(%2 : tensor<1x56x56x256xf32>) outs(%3 : tensor<1x56x56x256xf32>) {
+    ^bb0(%in: f32, %out: f32):
+      %5 = arith.addf %in, %out : f32
+      linalg.yield %5 : f32
+  } -> tensor<1x56x56x256xf32>
+  return %4 : tensor<1x56x56x256xf32>
+}
+
+// X = linalg.conv_2d_nhwc_hwcf
+// Y = linalg.conv_2d_nhwc_hwcf
+// add(X, Y)
+transform.sequence failures(propagate) {
+  ^bb0(%arg1: !pdl.operation):
+    %0 = transform.structured.match ops{["linalg.generic"]} in %arg1
+    %add = transform.structured.map_linalg_to_tpp filter{["tpp.add"]} in %0
+    %1 = get_producer_of_operand %add[0] : (!pdl.operation) -> !pdl.operation
+    %2 = get_producer_of_operand %add[1] : (!pdl.operation) -> !pdl.operation
+    // We assume both %1 and %2 to be conv operations.
+    %3 = merge_handles %1, %2 : !pdl.operation
+    %4 = transform.structured.pack %3 { blocking_factors = [32, 32] }
+   
+    // We need this to canonicalize pack(unpack)...
+    %func = transform.structured.match ops{["func.func"]} in %arg1 
+    // TODO: bubbling + canonicalization should be a proper transform op, at which point we will be
+    // able to preserve the handles and avoid rematching below.
+    transform.structured.packing_propagation %func
+
+    %5 = transform.structured.match ops{["linalg.generic"]} in %arg1
+    %add_packed = transform.structured.map_linalg_to_tpp filter{["tpp.add"]} in %5
+    %6, %loop:5 = transform.structured.fuse %add_packed { tile_sizes = [1, 1, 1, 1, 1] }
```

Comment:
which conv is this going to fuse with?

Reply:
both the relus get fused into the conv.

Reply:
relus?

Reply:
yes, relus. There are two relus and two convs. The consumer relu is fused into the conv,

Reply:
the example doesn't exist anymore, but it had two convs and an add, and that's fusing the add:
```
%add_packed = transform.structured.map_linalg_to_tpp filter{["tpp.add"]} in %5
%6, %loop:5 = transform.structured.fuse %add_packed { tile_sizes = [1, 1, 1, 1, 1] }
```
What am I missing?

Reply:
I see. I removed the example as I was not interesting. It was fusing the convs with the add. The other example fuses the relu into the conv.